raiden-ts/src/raiden.ts

```diff
@@ -1039,14 +1057,29 @@ export class Raiden {
     );
     const balance = await serviceTokenContract.functions.balanceOf(address);
 
-    assert(balance.gte(amount), `Insufficient token balance (${balance}).`);
+    assert(balance.gte(amount), `Insufficient token balance (${balance}).`, this.log.debug);
 
-    const approveTx = await serviceTokenContract.functions.approve(
-      userDepositContract.address,
-      depositAmount,
-    );
-    const approveReceipt = await approveTx.wait();
+    let approveTx;
+    try {
+      approveTx = await serviceTokenContract.functions.approve(
+        userDepositContract.address,
+        depositAmount,
+      );
+    } catch (err) {
+      this.log.error('Error sending approve tx', err);
+      throw err;
```

Comment:
Same here, please throw errors that can be localized and be explained to the user